### ISSUE TITLE
HLint fails in master

### ISSUE THREAD

**Issue by phadej** [ekmett/lens#1035]

**Comment by RyanGlScott** [comment]
For reference, [here](https://github.com/ekmett/lens/actions/runs/5475437015/jobs/9971537826#step:25:24) is the failure:

```
src/Control/Lens/Internal/FieldTH.hs:6:1-32: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Perhaps you should remove it.
Note: may require `{-# LANGUAGE TemplateHaskellQuotes #-}` adding to the top of the file

1 hint
Error: Process completed with exit code 1.
```

Which is warning about this code:

https://github.com/ekmett/lens/blob/25b8df2faa682f445359038966280a00b7083780/src/Control/Lens/Internal/FieldTH.hs#L3-L7

But this confuses me: the `{-# LANGUAGE TemplateHaskell #-}` line should only be reachable if using a very old version of GHC. Does `hlint` not take CPP into account?

**Comment by RyanGlScott** [comment]
It's possible that this is an occurrence of https://github.com/ndmitchell/hlint/pull/1522. This has been fixed in `hlint-3.6`, and indeed, I cannot reproduce the issue locally with that version of `hlint`. (I do encounter other `hlint` warnings, but those appear fixable.)
